src/tools/Misc.cpp

```diff
@@ -28,3 +28,8 @@ string misc::getPartNameFromReference(gsoap_resqml2_0_1::eml20__DataObjectRefere
 	return reference->ContentType.substr(reference->ContentType.rfind('=')+1) + "_" + reference->UUID + ".xml";
 }
 
+string misc::getPartNameFromReference(gsoap_eml2_2::eml22__DataObjectReference * reference)
```

Comment:
Yo could have factorized the code by creating a "private" template method called by both getPartNameFromReference.

Reply:
Did not factorise because of "UUID" vs "Uuid".

Reply:
Thanks